pallets/providers/README.md

```diff
@@ -1 +1,380 @@
+
+BSPs do not compete for user adoption as they do not offer distinctive services to one another and are not chosen by the user. Instead, they are assigned by the system, with considerations to ensure an even distribution of data.
+
+The remaining portion of the user's payment for storing a file, after the MSP's share, is evenly distributed among the assigned BSPs. This incentivizes BSPs to maintain the data they've committed to storing.
+
+### Sign Up Process
+
+The sign up process for Storage Providers is a two-step process to avoid malicious users from predicting the randomness used to generate the unique ID of the Storage Provider.
+
+1) The first step is the request to sign up, where the user provides the necessary information to become a Storage Provider, commiting to that information, and the necessary deposit is held.
+2) The second step is the confirmation of the sign-up, which must be done by the user (or a third-party) after enough time has passed to ensure that the randomness used to generate the unique ID was not predictable when the sign-up request (commitment) was made.
+
+This process exists because the unique ID of a Storage Provider is what determines if they can volunteer to store a new file of the system after a store request was made by a user, and if the randomness used to generate the unique ID was predictable, malicious users could generate multiple Storage Provider accounts with similar IDs and collude to store the same file, which would be detrimental to the system as it would allow file storage centralization, allowing censorship and data loss.
+
+### Sign Off Process
+
+The sign off process is simpler that the sign up process: the Storage Provider requests to sign off, and if it does not have any data currently in use (that means, no user file is currently being stored by this Storage Provider), the deposit is returned and the Storage Provider information is completely deleted from the system. This deletion is permanent, as the unique ID of the Storage Provider is generated using the runtime's randomness, and it is not possible to recreate it.
+
+### Capacity Management
+
+Storage Providers can change their capacity, increasing or decreasing it as they see fit. The new capacity has to be more than the minimum allowed by the runtime, more than the Storage Provider's used capacity, and the change is subject to a timelock to avoid spam attacks. The new deposit needed for the new capacity is calculated, and the user has to pay the difference if the new deposit is greater than the current deposit. If the new deposit is less than the current deposit, the held difference is returned to the user. This allows Storage Providers to adapt to the network's needs and their own infrastructure capabilities.
+
+## Extrinsics
```

Comment:
Everything passed this point, personally I would prefer it as crate level docs since you can then link storage and extrinsic functions directly. But I leave it up to you :)